src/FilterTracks.cxx

```diff
@@ -0,0 +1,133 @@
+# include "TrackPerf/FilterTracks.hxx"
+
+#include <math.h>
+
+#include <DD4hep/Detector.h>
+
+#include <EVENT/Track.h>
+#include <IMPL/LCCollectionVec.h>
+
+#include <marlin/AIDAProcessor.h>
+
+#include <AIDA/ITree.h>
+
+FilterTracks aFilterTracks ;
+
+FilterTracks::FilterTracks()
+  : Processor("FilterTracks")
+{
+  // modify processor description
+  _description = "FilterTracks processor filters a collection of tracks based on NHits and MinPt and outputs a filtered collection";
+
+  // register steering parameters: name, description, class-variable, default value
+  registerProcessorParameter("NHits",
+		  	     "Minimum number of hits on track",
+			     _NHits,
+			     _NHits
+			      );
+  
+  registerProcessorParameter("NHits1",
```

Comment:
Can you make these parameter names more descriptive? ie: `NHitsVertex`.

You can keep the numbers in the histogram class as they are.